api/server/server.go

```diff
@@ -260,52 +261,46 @@ func getContainersExport(eng *engine.Engine, version version.Version, w http.Res
 }
 
 func getImagesJSON(eng *engine.Engine, version version.Version, w http.ResponseWriter, r *http.Request, vars map[string]string) error {
-	if err := parseForm(r); err != nil {
+	var err error
+
+	if err = parseForm(r); err != nil {
 		return err
 	}
 
-	var (
-		err  error
-		outs *engine.Table
-		job  = eng.Job("images")
-	)
-
-	job.Setenv("filters", r.Form.Get("filters"))
-	// FIXME this parameter could just be a match filter
-	job.Setenv("filter", r.Form.Get("filter"))
-	job.Setenv("all", r.Form.Get("all"))
+	imagesConfig := graph.ImagesConfig{
+		Filters: r.Form.Get("filters"),
+		// FIXME this parameter could just be a match filter
+		Filter: r.Form.Get("filter"),
+		All:    r.Form.Get("all") == "1",
+	}
 
-	if version.GreaterThanOrEqualTo("1.7") {
-		streamJSON(job, w, false)
-	} else if outs, err = job.Stdout.AddListTable(); err != nil {
+	images, err := GetDaemon(eng).Repositories().CmdImages(&imagesConfig)
+	if err != nil {
 		return err
 	}
 
-	if err := job.Run(); err != nil {
-		return err
+	if version.GreaterThanOrEqualTo("1.7") {
+		return writeJSON(w, http.StatusOK, images)
 	}
 
-	if version.LessThan("1.7") && outs != nil { // Convert to legacy format
-		outsLegacy := engine.NewTable("Created", 0)
-		for _, out := range outs.Data {
-			for _, repoTag := range out.GetList("RepoTags") {
-				repo, tag := parsers.ParseRepositoryTag(repoTag)
-				outLegacy := &engine.Env{}
-				outLegacy.Set("Repository", repo)
-				outLegacy.SetJson("Tag", tag)
-				outLegacy.Set("Id", out.Get("Id"))
-				outLegacy.SetInt64("Created", out.GetInt64("Created"))
-				outLegacy.SetInt64("Size", out.GetInt64("Size"))
-				outLegacy.SetInt64("VirtualSize", out.GetInt64("VirtualSize"))
-				outsLegacy.Add(outLegacy)
+	legacyImages := []types.LegacyImage{}
```

Comment:
@jfrazelle this is similar to what we played with yesterday and probably can be removed (and I wouldn't mind) but I didn't make a fuss about this one because generating this legacy json was easy, while the one we were looking at yesterday was a pain.
If people want it removed, I can do so.... just need to hear some opinions.